[themintchoco] iP

themintchoco opened the pull request:
# Marshmallow

> Life is a marshmallow, easy to chew but hard to swallow." – Francis Bacon

Marshmallow frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn
- ~FAST~ _SUPER_ FAST to use

All you need to do is,
1. download it from [here]().
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

---

If you are a Java programmer, you can use it to practice Java too. Here's the `main`method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

themintchoco committed:
feat: Rename Duke to Marshmallow

themintchoco committed:
feat: Add ability to respond

themintchoco committed:
feat: Add echo functionality

themintchoco committed:
refactor: Command handling

themintchoco committed:
feat: Add tasks

themintchoco committed:
feat: Add ability to mark tasks as done

themintchoco committed:
feat: Add Events and Deadlines

themintchoco committed:
fix: Remove extraneous whitespace in Todo descriptions

themintchoco committed:
test: Add simple tests

themintchoco committed:
feat: Add exception handling

themintchoco committed:
feat: Add ability to delete tasks

themintchoco committed:
feat: Add persistence to disk

themintchoco committed:
Merge branch 'branch-Level-7'

themintchoco committed:
feat: Add datetime parsing

themintchoco committed:
Merge branch 'branch-Level-8'

themintchoco committed:
refactor: Extract classes to meet requirements

themintchoco committed:
refactor: Organize into packages to meet requirements

themintchoco committed:
Merge branch 'add-gradle-support'

themintchoco committed:
test: Add test for Parser and TaskList

themintchoco committed:
docs: Add JavaDoc

themintchoco committed:
style: Follow style guide

themintchoco committed:
feat: Add find command

themintchoco committed:
Merge branch 'branch-A-JavaDoc'

themintchoco committed:
Merge branch 'branch-A-CodingStandard'

themintchoco committed:
Merge branch 'branch-Level-9'

themintchoco committed:
fix: Fix regression in handling missing save file

josejhkim commented on src/main/java/marshmallow/storage/textfilestorage/TextFileStorage.java:
```diff
@@ -0,0 +1,76 @@
+ * @param <T> Type of items to save.
+ * @param <TF> Factory to construct items from save strings.
+ */
+public class TextFileStorage<T extends TextFileStorageSaveable,
```
I like how you used generics and interfaces!

josejhkim commented on src/main/java/marshmallow/Marshmallow.java:
```diff
@@ -0,0 +1,54 @@
+                    Command command = parser.parse(scanner);
+
+                    if (command == null) {
+                        ui.respond("I'm sorry, but I don't know what that means :(");
```
Perhaps it may be cleaner to have your "sorry" statement as a function in your Ui class, since ui.greet() is a separate function as well?

josejhkim commented on src/main/java/marshmallow/commands/ListTaskUnmarkCommand.java:
```diff
@@ -0,0 +1,21 @@
+import marshmallow.tasks.TaskList;
+import marshmallow.ui.Ui;
+
+public class ListTaskUnmarkCommand extends Command {
```
It'd be great to add JavaDoc comments for the different command classes to help with comprehension

josejhkim reviewed (commented):
Looks great! Overall great code quality, besides the (very) tiny nitpicks I mentioned

hyizhak commented on src/main/java/marshmallow/parser/Parser.java:
```diff
@@ -0,0 +1,78 @@
+     * @return Command representing the user input, or null if the input is invalid
+     */
+    public Command parse(Scanner s) {
+        String command = s.next();
```
If the user enters an empty line, s.next() will throw NoSuchElementException. Do you think you need to check if input exits before calling next()?

hyizhak commented on src/main/java/marshmallow/tasks/Deadline.java:
```diff
@@ -0,0 +1,35 @@
+     */
+    public Deadline(String description, String endTimeString) {
+        super(description);
+        this.endTime = LocalDateTime.parse(endTimeString, dtfParse);
```
If endTimeString is incorrectly formatted, LocalDateTime.parse() will throw a DateTimeParseException, which isn’t caught here. I think a try-catch block would help the robustness?

hyizhak commented on src/main/java/marshmallow/Marshmallow.java:
```diff
@@ -0,0 +1,54 @@
+    public void run() {
+        Parser parser = new Parser();
+
+        try (Scanner scanner = new Scanner(System.in)) {
```
Do you think this functionality can be implemented in the UI class so it interacts with the user more organizedly?

hyizhak reviewed (commented):
LGTM. Just a few nits to be fixed

themintchoco committed:
feat: Add GUI

themintchoco committed:
Merge branch 'branch-Level-10'

themintchoco committed:
fix: Add assertions for assumptions

Command handling methods rely on assumptions such as the existence of
the task list (or even the task itself).

We should document these assumptions using assertions.

Let's
* add assert statements to these methods

themintchoco committed:
refactor: Follow code quality guide

themintchoco committed:
Merge pull request #1 from themintchoco/branch-A-Assertions

Add assertions

themintchoco committed:
Merge branch 'master' into branch-A-CodeQuality

themintchoco committed:
Merge pull request #2 from themintchoco/branch-A-CodeQuality

Follow code quality guide

themintchoco committed:
feat: Simplify command syntax

At present, keywords (eg. todo, deadline) not only have to be written in
full but are also case sensitive.

This is not user friendly since case sensitivity is not meaningful and
the user should not have to type the entire command where there is no
ambiguity as to which command is desired.

Let's
* update the parser to be more flexible, by using case insensitive
  matching and allowing partial commands

themintchoco committed:
feat: Allow natural dates input

themintchoco committed:
Merge branch 'branch-C-NaturalDates'

themintchoco committed:
Merge branch 'branch-C-FriendlierSyntax'

themintchoco committed:
docs: Add JavaDoc

themintchoco committed:
fix: Fix regression preventing command handling

themintchoco committed:
feat: Add error handling for invalid input

themintchoco committed:
Merge branch 'branch-A-MoreErrorHandling'

themintchoco committed:
docs: Add user guide and README

themintchoco committed:
docs: Update README and add acknowledgements

themintchoco committed:
docs: Update Javadoc to align with guidelines, resolves #3

themintchoco committed:
feat: Add response when list is empty

themintchoco committed:
feat: Add case-insensitive filtering of task

themintchoco committed:
feat: Add response when filtered list is empty
